packages/saltcorn-builder/src/components/storage.js

```diff
@@ -147,7 +148,17 @@ export /**
             configuration={segment.configuration || {}}
           />
         );
-      } else if (segment.type === "action") {
+      } else if (segment.type === "page") {
```

Comment:
you can get this for free if you declare Page.craft.related.fields and segment_type, see https://github.com/saltcorn/saltcorn/blob/master/packages/saltcorn-builder/src/components/elements/ViewLink.js#L262 - then there is no need for a case in the storage functions - but only if we don't need name

Reply:
added fields and type to related property